adaptor/mongodb/client.go

```diff
@@ -189,6 +194,30 @@ func WithTail(tail bool) ClientOptionFunc {
 	}
 }
 
+func WithReadPreference(read_preference string) ClientOptionFunc {
+	return func(c *Client) error {
+		if read_preference == "" {
+			c.readPreference = DefaultReadPreference
+			return nil
+		}
+		switch read_preference {
```

Comment:
I'm thinking it'd be nice to do a `strings.ToLower(read_preference)` here so the preference is insensitive.